crud/src/main/java/org/openmbee/sdvc/crud/repositories/BaseElasticDAOImpl.java

```diff
@@ -1,24 +1,116 @@
 package org.openmbee.sdvc.crud.repositories;
 
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
 import java.util.Set;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.delete.DeleteRequest;
+import org.elasticsearch.action.delete.DeleteResponse;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.action.get.MultiGetItemResponse;
+import org.elasticsearch.action.get.MultiGetRequest;
+import org.elasticsearch.action.get.MultiGetResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.search.fetch.subphase.FetchSourceContext;
 import org.openmbee.sdvc.crud.config.DbContextHolder;
 import org.openmbee.sdvc.json.BaseJson;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+
+public abstract class BaseElasticDAOImpl {
 
-public abstract class BaseElasticDAOImpl implements BaseIndexDAO {
+    protected RestHighLevelClient client;
+    @Value("${elastic.limit.result}")
+    protected static int resultLimit;
+    @Value("${elastic.limit.term}")
+    protected static int termLimit;
+    protected static int readTimeout = 1000000000;
+    // :TODO save, saveAll --> updates have details of upsert method, should break out into helper method for create/update
+
+    @Autowired
+    public void setRestHighLevelClient(@Qualifier("clientElastic") RestHighLevelClient client) {
+        this.client = client;
+    }
 
     public String getIndex() {
         return DbContextHolder.getContext().getIndex();
     }
 
-    public Map<String, Object> findByIndexId(String indexId) {
+    public long count() {
+        // Returns the number of entities available.
+        return 0;
+    }
+
+    public void delete(BaseJson json) throws IOException {
+        // :TODO deletes by entity
+    }
+
+    public void deleteById(String index, String indexId) throws IOException {
+        client.delete(new DeleteRequest(index, null, indexId), RequestOptions.DEFAULT);
```

Comment:
i think all the type arguments still needs an actual type, elastic suggests just use "_doc"